app/controllers/payment-links/get-manage.controller.js

```diff
@@ -10,26 +10,25 @@ const { renderErrorView } = require('../../utils/response')
 const supportedLanguage = require('../../models/supported-language')
 const paths = require('../../paths')
 
-module.exports = (req, res) => {
+module.exports = async (req, res) => {
   const externalServiceId = req.service && req.service.externalId
   lodash.unset(req, 'session.editPaymentLinkData')
 
-  productsClient.product.getByGatewayAccountId(authService.getCurrentGatewayAccountId(req))
-    .then(products => {
-      const paymentLinks = products.filter(product => product.type === 'ADHOC')
-      const englishPaymentLinks = paymentLinks.filter(link => link.language === supportedLanguage.ENGLISH)
-      const welshPaymentLinks = paymentLinks.filter(link => link.language === supportedLanguage.WELSH)
-      const pageData = {
-        productsLength: paymentLinks.length,
-        englishPaymentLinks,
-        welshPaymentLinks,
-        paths,
-        externalServiceId
-      }
-      return response(req, res, 'payment-links/manage', pageData)
-    })
-    .catch((err) => {
-      logger.error(`[requestId=${req.correlationId}] Get ADHOC product by gateway account id failed - ${err.message}`)
-      renderErrorView(req, res)
-    })
+  try {
+    const products = await productsClient.product.getByGatewayAccountId(authService.getCurrentGatewayAccountId(req))
+    const paymentLinks = products.filter(product => product.type === 'ADHOC')
+    const englishPaymentLinks = paymentLinks.filter(link => link.language === supportedLanguage.ENGLISH)
+    const welshPaymentLinks = paymentLinks.filter(link => link.language === supportedLanguage.WELSH)
+    const pageData = {
+      productsLength: paymentLinks.length,
+      englishPaymentLinks,
+      welshPaymentLinks,
+      paths,
+      externalServiceId
+    }
+    return response(req, res, 'payment-links/manage', pageData)
+  } catch (err) {
+    logger.error(`[requestId=${req.correlationId}] Get ADHOC product by gateway account id failed - ${err.message}`)
+    return renderErrorView(req, res)
```

Comment:
This PR is great! So much clearer. 

Quick comment that now that we've set up error handling appropriately for the middleware architecture generic error handling can be done downstream of the controller (for custom error handling like redirecting the controller can take of those). 

The following would have the equivalent functionality but would be one less place the developer has to think about responding. 

```javascript
next(err)
```

Reply:
👍🏽 Updated in https://github.com/alphagov/pay-selfservice/pull/2471/files#diff-e3c9d7742c1ef0d275b4a361576dda5dd01c427955b6c94663e10f19dc739cd6R29